fonts.user.styl

```diff
@@ -50,85 +50,148 @@
-    n += ':not([id][style*="font-family"]):not([id][style*="font-family"] *)'
+// inputs, but not of button type #6
+append(composer, 'input:not([name=subjectbox]):not([type=button]):not([type=submit]):not([type=reset])');
 
-    // Exclude utility classes for fonts.
-    n += ':not(.monospace):not(.text-mono)'
+// Github search
+append(composer, '.QueryBuilder-StyledInputContainer')
 
-    // mathjax
-    n += ':not(mjx-math *)'
+// don't change the weight of headers and bold text
+append(bold, 'strong', 'b', '[class*=bold]', 'h1', 'h2', 'h3', 'h4', 'h5', 'h6');
+
+for split in split('|', join(' *|', icons)) {
+    append(icons, split);
+}
+
+for split in split('|', join(' *|', code)) {
+    append(code, split);
+}
+
+for split in split('|', join(' *|', composer)) {
+    append(composer, split);
+}
+
+for split in split('|', join(' *|', bold)) {
+    append(bold, split);
+}
 
-    // Feat: use mono font in input areas.
+// make sure to include all selectors that get different fonts int the :not
+:not({ unquote(join(', ', join(', ', icons), join(', ', code), join(', ', composer), 'body > div')) }) {
+    font-family: base       !important if base
+    font-style:  baseStyle  !important if baseStyle
+    letter-spacing: 0px     !important if letterSpacing
+}
+
+:not({ unquote(join(', ', bold)) }) {
+    font-weight: baseWeight            if baseWeight
+}
+
+{ unquote(join(', ', code)) } {
+    font-family: mono       !important if mono
+    font-style:  monoStyle  !important if monoStyle
+    font-weight: monoWeight            if monoWeight
+    letter-spacing: 0px     !important if letterSpacing
+}
+
+{ unquote(join(', ', composer)) } {
     if inputs {
-        // Exclude inputs of button type. #6
-        n += ':not(input:not([type=button])):not(textarea)'
-
-        // Use mono input font on Discord.
-        n += ':not([data-slate-editor=true] *)'
-        n += ':not([class*="channelTextArea-"] *)'
-
-        // Workaround for specificity issues.
-        :not(#x):not(#y):not(#z) {
-            // Discord.
-            [class*="channelTextArea-"],
-
-            // Gmail. #3
-            [contenteditable=true]:not([g_editable]) *,
-            input:not([name="subjectbox"]):not([type=button]),
-
-            textarea {
-                // Feat: set font options if values are not null.
-                font-family: mono       i if mono
-                font-style:  monoStyle  i if monoStyle
-                font-weight: monoWeight i if monoWeight
-            }
-        }
+        font-family: mono       !important if mono
+        font-style:  monoStyle  !important if monoStyle
+        font-weight: monoWeight            if monoWeight
+        letter-spacing: 0px     !important if letterSpacing
+    } else {
+        font-family: base       !important if base
+        font-style:  baseStyle  !important if baseStyle
+        font-weight: baseWeight            if baseWeight
+        letter-spacing: 0px     !important if letterSpacing
+    }
+}
```

Comment:
Wrap this in a `@-moz-document regexp('(file|http|(chrome|moz)-extension)://.+') { ... }` block to avoid injecting the entire metadata header. Also, remove `unquote` functions because they doesn't seem necessary.

Reply:
Oh I see, I'll have another look at the unquote, it did make a difference at some point so I kept them there.